### PR TITLE
[RFR]: Show notifications if available at mount

### PR DESCRIPTION
**Pull request by tkvw**
Notifications didn't show if there are any available before the `Notification` component is mounted.

### CODE REVIEW

**Comment by fzaninotto** [comment]
Can you explain in which case this can happen, i.e. which bug is it supposed to fix?

**Comment by tkvw** [comment]
Sure, user navigates to a restricted page without authorization -> with a custom saga I call `showNotification` -> user is redirected to `/login` -> `/login` has it's own `Notification` component so when it mounts, the notification in the redux-store will not render (because the initial state `open` is `false`). The `open` state should be recalculated on mount of the component (just like `componentWIllReceiveProps`)

File: packages/ra-ui-materialui/src/layout/Notification.js [hunk 1]
```diff
@@ -25,9 +25,16 @@ class Notification extends React.Component {
     state = {
         open: false,
     };
+    componentWillMount = () => {
+        this.setNextState();
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/Notification.js]: shouldn't you use `this.props` here?

Comment (tkvw) [reply packages/ra-ui-materialui/src/layout/Notification.js]: The args of `setNextState` defaults to `this.props`

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/layout/Notification.js]: True, I hadn't paid attention. Or rather, the code wasn't clear enough. Could you please pass `this.props` here and have no default value in `setNextState`?

File: packages/ra-ui-materialui/src/layout/Notification.js [hunk 2]
```diff
@@ -25,9 +25,16 @@ class Notification extends React.Component {
...
+    };
     componentWillReceiveProps = nextProps => {
+        this.setNextState(nextProps);
+    };
+
+    setNextState = ({ notification } = this.props) => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/Notification.js]: While we're at it, please rename this method to `setOpenState`

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**